docs/reference/providers/custom/building-custom-provider.md

```diff
@@ -0,0 +1,263 @@
+
+The CLI will also containerize and build your function code, passing the information about the built image to the deployment provider. Given this deployment specification, the provider can deploy whatever resources are needed. For example, given the function below:
+
+```ts
+import { api, bucket, faas } from '@nitric/sdk';
+
+const helloApi = api('main');
+
+const buck = bucket('images').for('writing');
+
+helloApi.get('/hello/:name', async (ctx: faas.HttpContext) => {
+  const { name } = ctx.req.params;
+
+  ctx.res.body = `Hello ${name}`;
+
+  return ctx;
+});
+```
+
+The CLI would convert it into the following resource spec:
+
+```json
+{
+  "spec": {
+    "resources": [
+      {
+        "name": "images",
+        "type": 2,
+        "Config": {
+          "Bucket": {}
+        }
+      },
+      {
+        "name": "main",
+        "Config": {
+          "Api": {
+            "Document": {
+              "Openapi": "{\"components\":{},\"info\":{\"title\":\"main\",\"version\":\"v1\"},\"openapi\":\"3.0.1\",\"paths\":{\"/hello/{name}\":{\"get\":{\"operationId\":\"hellonameget\",\"responses\":{\"default\":{\"description\":\"\"}},\"x-nitric-target\":{\"name\":\"hello\",\"type\":\"function\"}},\"parameters\":[{\"in\":\"path\",\"name\":\"name\",\"required\":true,\"schema\":{\"type\":\"string\"}}]}}}"
+            }
+          }
+        }
+      },
+      {
+        "name": "435cb11ee277e61657218d6ecb595e7a",
+        "type": 8,
+        "Config": {
+          "Policy": {
+            "principals": [
+              {
+                "name": "hello",
+                "type": 1,
+                "Config": null
+              }
+            ],
+            "actions": [2],
+            "resources": [
+              {
+                "name": "images",
+                "type": 2,
+                "Config": null
+              }
+            ]
+          }
+        }
+      },
+      {
+        "name": "hello",
+        "type": 1,
+        "Config": {
+          "ExecutionUnit": {
+            "Source": {
+              "Image": {
+                "uri": "custom-provider-demonstration-hello"
+              }
+            },
+            "workers": 1,
+            "type": "default"
+          }
+        }
+      }
+    ]
+  },
+  "attributes": {
+    "project": "custom-provider-demonstration",
+    "stack": "custom-stack"
+  }
+}
+```
+
+The spec contains a list of resources and global attributes that can be passed to the deployment provider. The list of resources for this deployment contains an API, bucket, and execution unit with the information required to configure them for the cloud. The default project and stack attributes are also provided. You will notice in the execution unit config the image is provided with a URI pointing to the local image name.
+
+### Setting up the server
+
+The deployment provider requires an implementation for deploying and a method for destroying the cloud resources, called `Up` and `Down` respectively.
```

Comment:
```suggestion
The deployment provider requires implementations for deploying and destroying cloud resources, called `Up` and `Down` respectively.
```

Reply:
If the commands are in quotes, probably should match the case - 'up' , 'down'